clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/Estimator.java

```diff
@@ -0,0 +1,78 @@
+package com.linkedin.venice.datarecovery;
+
+import static java.lang.Thread.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * DataRecoveryExecutor is the engine to run tasks in data recovery.
+ */
+public class Estimator {
+  private final Logger LOGGER = LogManager.getLogger(Estimator.class);
+  private final static int DEFAULT_POOL_SIZE = 10;
+  private final static int DEFAULT_POOL_TIMEOUT_IN_SECONDS = 30;
+  private final int poolSize;
+  private final ExecutorService pool;
+  private List<PlanningTask> tasks;
+
+  public Estimator() {
+    this(DEFAULT_POOL_SIZE);
+  }
+
+  public Estimator(int poolSize) {
+    this.poolSize = poolSize;
+    this.pool = Executors.newFixedThreadPool(this.poolSize);
+  }
+
+  public void perform(Set<String> storeNames, EstimateDataRecoveryTimeCommand.Params params) {
+    tasks = buildTasks(storeNames, params);
+    List<CompletableFuture<Void>> taskFutures = tasks.stream()
+        .map(dataRecoveryTask -> CompletableFuture.runAsync(dataRecoveryTask, pool))
+        .collect(Collectors.toList());
+    taskFutures.stream().map(CompletableFuture::join).collect(Collectors.toList());
+    displayTaskResult();
+
+    shutdownAndAwaitTermination();
+  }
+
+  private void shutdownAndAwaitTermination() {
+    pool.shutdown();
+    try {
+      if (!pool.awaitTermination(DEFAULT_POOL_TIMEOUT_IN_SECONDS, TimeUnit.SECONDS)) {
+        // Cancel currently executing tasks.
+        pool.shutdownNow();
+      }
+    } catch (InterruptedException e) {
+      currentThread().interrupt();
+    }
+  }
```

Comment:
Alyssa, with the new change (https://github.com/linkedin/venice/commit/3a14ee71c1cb54ba60b0c6ed59d959c629df6133), many codes in this class can be removed, because I moved the command part into the base class `DataRecoveryWorker`. Take a look at it (DataRecoveryMonitor is an example), probably only function `buildTasks` and `displayTaskResult` need an override. Other functions can simply be removed.